mws/resource_mws_budget.go

```diff
@@ -0,0 +1,149 @@
+		return m
+	})
+	p := common.NewPairID("account_id", "budget_configuration_id")
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var create billing.CreateBudgetConfigurationBudget
+			common.DataToStructPointer(d, s, &create)
+			for i := range create.AlertConfigurations {
+				var err error
+				create.AlertConfigurations[i].QuantityThreshold, err = StringToBigDecimal(create.AlertConfigurations[i].QuantityThreshold)
+				if err != nil {
+					return err
+				}
+			}
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			budget, err := acc.Budgets.Create(ctx, billing.CreateBudgetConfigurationRequest{Budget: create})
+			if err != nil {
+				return err
+			}
+			d.Set("budget_configuration_id", budget.Budget.BudgetConfigurationId)
+			d.Set("account_id", c.Config.AccountID)
+			common.StructToData(budget.Budget, s, d)
+			p.Pack(d)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			_, id, err := p.Unpack(d)
+			if err != nil {
+				return err
+			}
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			budget, err := acc.Budgets.GetByBudgetId(ctx, id)
+			if err != nil {
+				return err
+			}
+			for i := range budget.Budget.AlertConfigurations {
+				budget.Budget.AlertConfigurations[i].QuantityThreshold, err = BigDecimalToString(budget.Budget.AlertConfigurations[i].QuantityThreshold)
+				if err != nil {
+					return err
+				}
+			}
+			return common.StructToData(budget.Budget, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var update billing.UpdateBudgetConfigurationBudget
+			_, id, err := p.Unpack(d)
+			if err != nil {
+				return err
+			}
+			common.DataToStructPointer(d, s, &update)
+			for i := range update.AlertConfigurations {
+				update.AlertConfigurations[i].QuantityThreshold, err = StringToBigDecimal(update.AlertConfigurations[i].QuantityThreshold)
+				if err != nil {
+					return err
+				}
+			}
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			budget, err := acc.Budgets.Update(ctx, billing.UpdateBudgetConfigurationRequest{
+				Budget:   update,
+				BudgetId: id,
+			})
+			if err != nil {
+				return err
+			}
+			return common.StructToData(budget.Budget, s, d)
+		},
+		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			_, id, err := p.Unpack(d)
+			if err != nil {
+				return err
+			}
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			return acc.Budgets.DeleteByBudgetId(ctx, id)
+		},
+		Schema: s,
+	}
+}
+
+// API always return a BigDecimal for QuantityThreshold in string representation.
+// To avoid drift, we convert the provided number to BigDecimal in string representation
+// when making requests to the API.
```

Comment:
Have to change this logic to instead use `SetCustomSuppressDiff` for `alert_configurations.#.quantity_threshold`